### DISCUSSION TITLE
SnipeIT Frontend Enhancement for shown lists

### DISCUSSION THREAD

**Discussion by thomasschulteimbus** [snipe/snipe-it#10338] [Ideas]
Roadmap Question

Different People looking at the application from different angles need to have different list elements
Also different assets with different custom fields attached to the asset (B.t.w a really greetd feature of snipeIT) need different lists.

Question 1: Would it be possible to let us preconfigure views per asset type and combine that with specified filters to get with less clicks the information needed to work?
Question 2: Would it also be possible to give each user a "Save this View Configuration under this name" and a "method to select and show the saved view"?

**Comment by bkibbeyWork** [comment]
We have a similar set of features in the MDM we use for MacBooks... Addigy.  It works like this... 

Each table view has some ability to set Columns AND Filters.
Privileged users can save their current set of columns and filters as a "View".
Any user can select one of the saved Views to bring up that set of saved columns and filters.

**Comment by tony-pinto** [comment]
I would like to hide those "areas" that I duo not use - for example if I am only using this tool for tracking licenses but not other assets, consumables etc. I would like to be able to hide those features in the front end of this of this tool that I am not using. Is this possible ? I whave not been able to find views to do this.

**Reply by cswinton147** [reply]
why not just disable them in groups

**Comment by tony-pinto** [comment]
How do I do that and will they still show up in the dashboard ? I just want them to not show up in the dashboard

**Reply by cswinton147** [reply]
you do this through groups under admin

those items wont show up in the left hand menu.

![image](https://user-images.githubusercontent.com/97762891/149586660-b6f4f82f-d6ab-4e3c-b583-a98b96898a77.png)

**Comment by tony-pinto** [comment]
Thank you !  is there any way to hide the corresponding buttons which show up in the dashboard ?

**Reply by snipe** [reply]
There are two different problems being discussed here. Limiting permissions is useful only for users who are not superadmins, and those users don’t have access to the dashboard in the first placement so removing permissions won’t help hide things in the sidenav or the dashboard for superadmins. 

The syntax we use is `@can`, which only implies permissions and not whether or not you *want* things visible.

**Comment by tony-pinto** [comment]
I am  primarily looking for a way to hide buttons in the dashboard and the menu items by the super admin (from anyone) which are not used for tracking. So,  it comes down to customizing this product which I find the most useful from the perspective of it's functionality and security focused design. 

I am interested in tracking only assets and software licenses but not consumables etc. and I suspect that in the wide world out there, there are others who might want to only track one or two things on the dashboard and menu, so there is a need to hide those elements not being used (so this is a feature request).  I hope this gives an idea about what I am asking about. If there is a way to easily do what I am asking in a future release, that would be great.

**Reply by snipe** [reply]
@tony-pinto please try to keep replies within the same thread within the discussion. It gets pretty confusing otherwise. 

I understand the ask, I’m just saying that the solution (permissions) will not work as you expect it to, since we do not look at whether you WANT to display it, just whether you CAN see it. 

This topic has come up before, and it’s not on the immediate roadmap.

**Comment by tony-pinto** [comment]
Thank you for you reply and sorry as I thought I was replying in the same thread

**Comment by bkibbeyWork** [comment]
Seeing the replies here reminded me about this.  

Here is an example of the Addigy tool we use, that allows for different views for table outputs.  I would LOVE it if SnipeIT had something similar.  It is very flexible and handy to use for sorting/ finding things.

![image](https://user-images.githubusercontent.com/4581975/149630089-1a159a7c-b063-442a-9689-61360e8f404b.png)
![image](https://user-images.githubusercontent.com/4581975/149630107-70b33d21-8fac-4b31-aa6c-fda88ed42c9a.png)
![image](https://user-images.githubusercontent.com/4581975/149630115-c158357c-0786-4fc9-a3d3-ae7fdebfcf06.png)
![image](https://user-images.githubusercontent.com/4581975/149630119-79db02bd-4098-4464-9f13-7ba5b7f0a9a9.png)
![image](https://user-images.githubusercontent.com/4581975/149630125-4ba07896-e20e-4fac-a59d-f2f503791fdd.png)

**Reply by snipe** [reply]
We are constrained by the data table library we use.
